From 5ff76fae87a9fe10ddaef224c3f19ad9e0dd7de8 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 10 Mar 2022 16:44:45 -0300
Subject: [PATCH 038/142] timezone: Fix clang operator precedence warning

clang issues:

zdump.c:669:7: error: operator '?:' has lower precedence than '&'; '&'
will be evaluated first [-Werror,-Wbitwise-conditional-parentheses]
                    ? (delta(&tm, &lotm) == t - lot

which is exactly what the code intendes.  Instead of adding a
warning suppress, place parentheses around the '&' expression.
---
 timezone/zdump.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/timezone/zdump.c b/timezone/zdump.c
index b532fe3eae..bfc01faffa 100644
--- a/timezone/zdump.c
+++ b/timezone/zdump.c
@@ -665,7 +665,7 @@ hunt(timezone_t tz, char *name, time_t lot, time_t hit)
 		else if (t >= hit)
 			--t;
 		tm_ok = my_localtime_rz(tz, &t, &tm) != NULL;
-		if (lotm_ok & tm_ok
+		if ((lotm_ok & tm_ok)
 		    ? (delta(&tm, &lotm) == t - lot
 		       && tm.tm_isdst == lotm.tm_isdst
 		       && strcmp(abbr(&tm), ab) == 0)
-- 
2.34.1

